### PR TITLE
Speed-up-vm-fetching

### PR DESCRIPTION
**Pull request by jecisc**
Speedup vm fetching.

Instead of ensuring the rights of each executables when creating the vms, I now create all the VM then ensure they are executables in one command. 

This also fix a bug. If we updated a VM and already had the executable file in cache, it did not ensured that the new executable was executable.

### CODE REVIEW

File: src/PharoLauncher-Spec2/PhLVMPresenter.class.st [hunk 1]
```diff
@@ -96,9 +96,8 @@ PhLVMPresenter >> initializePresenters [
 		contextMenuFromCommandsGroup: [ self rootCommandsGroup / 'Context Menu' ];
 		beResizable;
 		beMultipleSelection;
-		yourself.
-	"Run in a separate process to do not block UI. Can take some seconds."
-	[ vmTable items: self datasource virtualMachines ] fork.
```
Comment (demarey) [inline src/PharoLauncher-Spec2/PhLVMPresenter.class.st:101]: just a question about the previous line: is the speedup enough to do not let the user feel that the UI is "blocked"?

File: src/PharoLauncher-Core/PhLVirtualMachineManager.class.st [hunk 2]
```diff
@@ -294,6 +294,23 @@ PhLVirtualMachineManager >> downloadPathOSTypeSegment [
 
 ]
 
+{ #category : #querying }
+PhLVirtualMachineManager >> ensureExecutionRightsOf: aCollectionOfVms [
```
Comment (demarey) [inline src/PharoLauncher-Core/PhLVirtualMachineManager.class.st]: very good idea to fix rigths of all VMs in a single command!

**Review by demarey — COMMENTED** [review]
thanks for the PR!

File: src/PharoLauncher-Spec2/PhLVMPresenter.class.st (hunk 1 above)
Comment (jecisc) [reply src/PharoLauncher-Spec2/PhLVMPresenter.class.st:101]: I can profile tomorrow to know the exact time it takes to open. 
Good thing is that now the number of vm will probably not influence :)

**Pull request merged**